libindy_vdr/src/ffi/pool.rs

```diff
@@ -40,6 +41,9 @@ pub static POOLS: Lazy<RwLock<BTreeMap<PoolHandle, PoolInstance>>> =
 pub static POOL_CACHE: Lazy<RwLock<Option<Arc<dyn PoolTransactionsCache>>>> =
     Lazy::new(|| RwLock::new(Some(Arc::new(InMemoryCache::new()))));
 
+pub static LEDGER_TXN_CACHE: Lazy<RwLock<Option<Cache<String, (String, RequestResultMeta)>>>> =
```

Comment:
I think there may be an issue with having one global cache. Clients often connect to multiple ledgers, and the caches need to be distinct, so that a schema ID from one ledger doesn't resolve via the cache when requesting from another.

Reply:
the cache key includes the ledger ID:
![image](https://github.com/hyperledger/indy-vdr/assets/36937407/679a740d-4e88-49e3-8777-07f23b9a0116)

Reply:
I'm using sha256 to hash the key so we can save space in the cache. We shouldn't have any conflict caching for multiple ledgers in the same cache

Reply:
I believe that `ledgerId` identifies the plenum 'subledger': 0 is the pool, 1 is the domain ledger for most commonly used transactions, 2 is the config ledger. It is good to include that in the cache key, but it won't differentiate responses from (say) CANdy and Sovrin.

Reply:
how did you differentiate between ledgers when you created the pool cache? or did you not need to?

Reply:
It's not needed there, the set of genesis transactions (the cached values) are enough to differentiate them.

Reply:
could we use the pool merkle tree as part of the key? that way it would differentiate between each pool? Having a global cache is nice because then we just have one cache location on the file system, I'd rather not make the user maintain multiple db files for each ledger they use

Reply:
You could, but then you'd lose the cache when the pool transactions were updated. I think even if you picked like, the first 4 transactions you would risk strange effects when connecting to two ledgers that had forked off of the same base (although that is somewhat theoretical).

Reply:
hmm, is there any way to consistently ID a pool across pool refreshes/updates?

Reply:
if not I could add a pool id to the type that we could leverage in our caching